### ISSUE TITLE
Allowed casting for classical types is non-transitive

### ISSUE THREAD

**Issue by ajberdy** [openqasm/openqasm#527]
### What is the file name(s) where you found the bug?

types.rst

### What is the current text that needs to be fixed or removed? (You can copy and paste)

[Allowed Casts table](https://openqasm.com/language/types.html#allowed-casts)

### What should the documentation say instead?

Fixed table

### What content needs to be updated or replaced? (You can copy and paste)

[Allowed Casts table](https://openqasm.com/language/types.html#allowed-casts)

**Comment by jwoehr** [comment]
For background, see #268 and specifically @mbhealy 's [comment](https://github.com/openqasm/openqasm/pull/268#issuecomment-914358848)

**Comment by blakejohnson** [comment]
Right, this was an intentional choice. We wanted users to explicitly write `angle(bit(int_var))` if they want the integer to be interpreted as a bucket of bits, or `angle(float(int_var))` if it should be interpreted like a floating-point value.

**Comment by blakejohnson** [comment]
Let's examine the behavior of integer literals.

**Comment by blakejohnson** [comment]
In TSC meeting today, we noticed that the spec has many examples of implicit `int` literal to `angle` conversions (for example, see the `stdgates.inc` definitions of `id`, `u1`, and `u2` which use the literal `0` as the parameter to a gate). After some discussion, we resolved to allow implicit conversion of `int` to `angle` under the assumption that we take the path `int` -> `float` -> `angle`.

This doesn't really address transitivity of OpenQASM's type casting, but it will fix the specific issue around use of `angle`s.

A separate issue will be opened to track this change.

**Comment by levbishop** [comment]
The specific case of `angle` zero is separated into #594

**Comment by LestlinRobins** [comment]
Hi! I’d be happy to take this up and submit a PR for the documentation updates. If possible, could you assign this issue to me? Thanks!

**Comment by jwoehr** [comment]
I think this is already being covered in #666
